drivers/postgres/postgres.go

```diff
@@ -228,42 +210,28 @@ ORDER BY pt.tgrelid
 			table.Triggers = triggers
 		}
 
-		// columns comments
-		columnCommentRows, err := p.db.Query(`
-SELECT pa.attname AS column_name, pd.description AS comment
-FROM pg_stat_all_tables AS ps ,pg_description AS pd ,pg_attribute AS pa
-WHERE ps.relid=pd.objoid
-AND pd.objsubid != 0
-AND pd.objoid=pa.attrelid
-AND pd.objsubid=pa.attnum
-AND ps.relname = $1
-AND ps.schemaname = $2`, tableName, tableSchema)
-		defer columnCommentRows.Close()
-		if err != nil {
-			return errors.WithStack(err)
-		}
-
-		columnComments := make(map[string]string)
-		for columnCommentRows.Next() {
-			var (
-				columnName    string
-				columnComment string
-			)
-			err = columnCommentRows.Scan(&columnName, &columnComment)
-			if err != nil {
-				return errors.WithStack(err)
-			}
-			columnComments[columnName] = columnComment
-		}
-
 		// columns
 		columnRows, err := p.db.Query(`
-SELECT column_name, column_default, is_nullable, data_type, udt_name, character_maximum_length
-FROM information_schema.columns
-WHERE table_name = $1
-AND table_schema = $2
-ORDER BY ordinal_position
-`, tableName, tableSchema)
+SELECT
+    attr.attname AS column_name,
+    pg_get_expr(def.adbin, def.adrelid) AS column_default,
+    NOT (attr.attnotnull OR tp.typtype = 'd' AND tp.typnotnull) AS is_nullable,
+    CASE
+        WHEN attr.atttypid::regtype = ANY(ARRAY['character varying'::regtype, 'character varying[]'::regtype]) THEN
```

Comment:
fmfm.

Reply:
Yeah, I’m not that happy with this part. Looking over pg_type again this should really be a test for

```sql
    attr.atttypid::regtype = 'character varying'::regtype
OR (tp.typarray != 0 AND tp.typarray::regtype = 'character varying'::regtype)
```

Or, just drop the whole notion of transforming `character varying` to `varchar` altogether ;-)